src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = this.sidesArray.splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
+
+  protected semiperimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Triangle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    readonly shape: Shape = Shape.triangle,
+  ) {
+    if (this.sidesArray.some((side) => side <= 0)) {
+      throw new Error('length of some side is <= 0');
+    }
 
-export class Rectangle implements Figure {}
+    if (this.isSidesWrong) {
+      throw new Error(
+        // eslint-disable-next-line max-len
+        'Sides`s values are wrong (the longest side of a triangle is >= than a sum of two others)',
+      );
+    }
+  }
+
+  getArea(): number {
+    return +Math.sqrt(
+      this.semiperimeter *
+        (this.semiperimeter - this.a) *
+        (this.semiperimeter - this.b) *
+        (this.semiperimeter - this.c),
+    ).toFixed(2);
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    readonly shape: Shape = Shape.circle,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius is wrong (radius is <= 0)');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+    readonly shape: Shape = Shape.rectangle,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error(
+        'Sides`s values are wrong (one of side`s values is <= 0)',
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. Consider rephrasing the error message to be more descriptive, such as 'Rectangle side lengths must be greater than 0'.